webapp/src/shared/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,52 @@
+import { ReactComponent as ArrowBack } from './assets/icons/arrow-back.svg';
+import { ReactComponent as ArrowForward } from './assets/icons/arrow-forward.svg';
+import { ReactComponent as Chat } from './assets/icons/chat.svg';
+import { ReactComponent as Edit } from './assets/icons/edit.svg';
+import { ReactComponent as File } from './assets/icons/file.svg';
+import { ReactComponent as Login } from './assets/icons/login.svg';
+import { ReactComponent as Logout } from './assets/icons/logout.svg';
+import { ReactComponent as Play } from './assets/icons/play.svg';
+import { ReactComponent as Search } from './assets/icons/search.svg';
+import { ReactComponent as Users } from './assets/icons/users.svg';
+
+type IconType =
+  | 'ARROW_BACK'
+  | 'ARROW_FORWARD'
+  | 'CHAT'
+  | 'EDIT'
+  | 'FILE'
+  | 'LOGIN'
+  | 'LOGOUT'
+  | 'PLAY'
+  | 'SEARCH'
+  | 'USERS';
+
+type SVGProps = React.SVGProps<SVGSVGElement> & {
+  title?: string | undefined;
+};
+
+type SVGComponent = React.FunctionComponent<SVGProps>;
+
+const Icons: { [key in IconType]: SVGComponent } = {
+  ARROW_BACK: ArrowBack,
+  ARROW_FORWARD: ArrowForward,
+  CHAT: Chat,
+  EDIT: Edit,
+  FILE: File,
+  LOGIN: Login,
+  LOGOUT: Logout,
+  PLAY: Play,
+  SEARCH: Search,
+  USERS: Users,
+};
+
+interface IconProps {
+  type: IconType;
+  size?: string | number | undefined;
+  color?: string | undefined;
```

Comment:
@vdedios These articles explain it better than I could:
https://react-typescript-cheatsheet.netlify.app/docs/basic/troubleshooting/types/#enum-types
https://www.executeprogram.com/blog/typescript-features-to-avoid

Reply:
Ok I see your point!! I'd say it is a good fit for the ColorType but what about the numeric values for the IconSize? I think we need an enum for that. We can make it constant to avoid passing any numeric value (https://fettblog.eu/tidy-typescript-avoid-enums/#prefer-union-types)

Reply:
I've seen you already solved the IconSizes passing it as string but I still believe using an enum for it makes more sense. Passing just a value does not give much semantic meaning (is it small, is it large?) and, also, imagine we want to update the sizes of our icon because we realize that the size of `24px` is too small. We'll have to update everywere we used that size to the new value which is painfull. With an enum you only have to update the value in one place...

I'd say using union string types makes sense for color **denominations** because they are not very prone to change, but for sizes, color codes etc. makes more sense to use variables.

Reply:
Thank you, good points. I've made the change :+1: